contracts/src/LightClient.sol

```diff
@@ -182,27 +182,31 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     /// @dev Initialization of contract variables happens in this method because the LightClient
     /// contract is upgradable and thus has its constructor method disabled.
     /// @param _genesis The initial state of the light client
-    /// @param _genesisStakeTableState The initial stake state of the light client
+    /// @param _genesisStakeTableState The initial stake table state of the light client
     /// @param _stateHistoryRetentionPeriod The maximum retention period (in seconds) for the state
     /// history
     function _initializeState(
         LightClientState memory _genesis,
         StakeTableState memory _genesisStakeTableState,
         uint32 _stateHistoryRetentionPeriod
     ) internal {
-        // stake table commitments and threshold cannot be zero, otherwise it's impossible to
-        // generate valid proof to move finalized state forward.
-        // Whereas blockCommRoot can be zero, if we use special value zero to denote empty tree.
-        // feeLedgerComm can be zero, if we optionally support fee ledger yet.
+        // The viewNum and blockHeight in the genesis state must be zero to indicate that this is
+        // the
+        // initial state. Stake table commitments and threshold cannot be zero, otherwise it's
+        // impossible to
+        // generate valid proof to move finalized state forward. The stateHistoryRetentionPeriod
+        // must be
+        // at least 1 hour to ensure proper state retention.
         if (
             _genesis.viewNum != 0 || _genesis.blockHeight != 0
                 || BN254.ScalarField.unwrap(_genesisStakeTableState.blsKeyComm) == 0
                 || BN254.ScalarField.unwrap(_genesisStakeTableState.schnorrKeyComm) == 0
                 || BN254.ScalarField.unwrap(_genesisStakeTableState.amountComm) == 0
-                || _genesisStakeTableState.threshold == 0
+                || _genesisStakeTableState.threshold == 0 || _stateHistoryRetentionPeriod < 1 hours
```

Comment:
Can we define a constant for the minimum period of `1 hours` and then use that here and in `setstateHistoryRetentionPeriod`?

Reply:
sure, we can define a constant. Historically, I've been trying to minimize the use of constants in upgradeable contracts to avoid storage layout issues (even though it's stored in the contract's bytecode, if the same variable is no longer a constant in a future version of the contract, it would mess up the storage layout). However, since we use the inheritance pattern for handling upgrades, we shouldn't run into that issue. Will proceed

Reply:
we spoke offline and agreed not to set a constant since constants can't be upgraded and this constant is one that does have the possibility of changing. Instead, extra comments were added to ensure that developers and users are aware of the min state history retention period https://github.com/EspressoSystems/espresso-sequencer/commit/58f43df708454d978a40e7614f154abd819ecdb4